crates/biome_markdown_parser/src/lexer/mod.rs

```diff
@@ -181,10 +179,31 @@ impl<'src> MarkdownLexer<'src> {
         match dispatched {
             WHS => self.consume_newline_or_whitespace(),
             MUL | MIN | IDT => self.consume_thematic_break_literal(),
+            HAS => self.consume_header(),
             _ => self.consume_textual(),
         }
     }
 
+    fn consume_header(&mut self) -> MarkdownSyntaxKind {
+        self.assert_at_char_boundary();
+
+        let mut level = 0;
+        while matches!(self.current_byte(), Some(b'#')) {
+            self.advance(1);
+            level += 1;
+        }
+
+        match level {
+            1 => MD_HEADER1,
+            2 => MD_HEADER2,
+            3 => MD_HEADER3,
+            4 => MD_HEADER4,
+            5 => MD_HEADER5,
+            6 => MD_HEADER6,
+            _ => ERROR_TOKEN,
+        }
+    }
```

Comment:
Now that I'm looking at this again, this isn't right. The purpose of the lexer is to convert source code into an "alphabet" of sorts for the parser to interpret into a syntax tree. So it should output the "leaf" token, which would be the token for `#`, which you can get the SyntaxKind for via the macro `T![#]`.

I recommend taking a look at how the HTML lexer works, particularly how `consume_token` works here:

https://github.com/biomejs/biome/blob/766492f3535aef3a0a9bef47d607d07f8b70c749/crates/biome_html_parser/src/lexer/mod.rs#L54-L64

Reply:
Alternatively, another approach here would be to make each header it's own token. In the ungram, it could look like this: 

```
MdHeader1 = '#'
MdHeader2 = '##'
MdHeader3 = '###'
etc.
```

And then what you currently have here would make more sense. I'm not sure if I would recommend going that route though.

Reply:
with your proposal of grammar change:

```
thread 'main' panicked at /.cargo/registry/src/index.crates.io-6f17d22bba15001f/quote-1.0.38/src/runtime.rs:490:9:
"##_token" is not a valid Ident
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

Reply:
Yeah, in order to do that you have to update some of the codegen code. I would recommend doing what my first says.